src/guide/migration/emits-option.md

```diff
@@ -0,0 +1,94 @@
+---
+title: emits Option
+badges:
+  - new
+---
+
+# `emits` Option <MigrationBadges :badges="$frontmatter.badges" />
+
+## Overview
+
+Vue 3 now offers an `emits` option similar to the existing `props` option. This option can be used to define the events that a component can emit to its parent.
+
+## 2.x Behavior
+
+In Vue 2, you can define the props that a component received, but you can't declare which events it can emit:
+
+```html
+<template>
+  <p>{{ text }}</p>
+  <button v-on:click="$emit('accepted')">OK</button>
+</template>
+<script>
+  export default {
+    props: ['text']
+  }
+</script>
+```
+
+## 3.x Behavior
+
+Similar to props, the events that the component emit can now be defined with the `emits` option.
+
+```javascript
+<template>
+  <p>{{ text }}</p>
+  <button v-on:click="$emit('accepted')">OK</button>
+</template>
+<script>
+  export default {
+    props: ['text'],
+    emits: ['accepted']
+  }
+</script>
+```
+
+The option also accepts an object notation, which allows the developer to define validators for the arguments that are passed with the emitted event, similar to validators in props definitions.
+
+For more information on this, please read the [API documentation for this feature](../api/options-data.md#emits).
+
+## Migration Strategy
+
+It is highly recommended that you document all of the emitted events of your components this way because of the [removal of the `.native` modifier](./native-modifier-removed.md).
+
+All events not defined with `emits` are now added as DOM event listeners to the components root node (unless `inheritAttrs: false` has been set).
+
+### Examnple
+
+If ou have components that re-emit native events to their parent, this would now lead to two events being fired:
+
+```html
+<template>
+  <button v-on:click="$emit('click', event)">OK</button>
+</template>
+<script>
+  export default {
+    props: ['text'],
+    emits: [] // without declared event
+  }
+</script>
+```
+
+When a parent listens for the `click` event on the component:
+
+```html
+<my-button v-on:click="handleClick"></my-button>
+```
+
+it would now be triggered _twice_:
+
+- Once from `$emit()`
+- Once from a native event listener applied to the root element
+
+Here you have two options:
+
+1. Properly declare the `click` event. This is useful if you actually do add some logic to that event handler in `<my-button>`
+2. remove the re-emitting of the event, since the parent can now listen for the native event easily, without adding `.native`. Suitable when you really only re-emit the event anyway.
```

Comment:
```suggestion
2. Remove the re-emitting of the event, since the parent can now listen for the native event easily, without adding `.native`. Suitable when you really only re-emit the event anyway.
```